README.md

```diff
@@ -57,25 +63,35 @@ jobs:
   clang-tidy:
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v2
-      - name: Build project and/or unit tests
-        run: ./your-build-script.sh
-      - name: Run clang-tidy
-        run: ./your-clang-tidy-script.sh --fixes-path fixes.yaml
-      - name: Run clang-tidy-pr-comments action
-        uses: platisd/clang-tidy-pr-comments@master
-        with:
-          # The GitHub token (or a personal access token)
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          # The path to the clang-tidy fixes generated previously
-          clang_tidy_fixes: fixes.yaml
-          # Optionally set to true if you want the Action to request
-          # changes in case warnings are found
-          request_changes: true
-          # Optionally set the number of comments per review
-          # to avoid GitHub API timeouts for heavily loaded
-          # pull requests
-          suggestions_per_comment: 10
+    - uses: actions/checkout@v2
+    - name: Install clang-tidy
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y clang-tidy
+    - name: Prepare compile_commands.json
+      run: |
+        cmake -B build -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+    - name: Create results directory
+      run: |
+        mkdir clang-tidy-result
+    - name: Analyze
+      run: |
+        git diff -U0 HEAD^ | clang-tidy-diff -p1 -path build -export-fixes clang-tidy-result/fixes.yml
```

Comment:
What does this do? Run `clang-tidy-diff` only for the files changed in the last commit or something? Needs more explanation if we are to add it to the README.

Reply:
> Run clang-tidy-diff only for the files changed in the last commit or something?

Yes, exactly. To not waste time running the `clang-tidy` on unchanged files. Very handy for pull requests.

Reply:
What happens when a pull request gets created containing multiple commits? Won't that limit the analysis only to the files modified in the last commit?

Also, since the Action is taking care of that, I'd say let's go for simplicity instead of speed/optimization.

Reply:
> What happens when a pull request gets created containing multiple commits? Won't that limit the analysis only to the files modified in the last commit?

This depends on the diff you pass to it. I use the following syntax:

```
git diff -U0 HEAD^ | clang-tidy-diff -p1 -path build -export-fixes clang-tidy-result/fixes.yml
```

```HEAD^``` means "first parent of the current branch", therefore, I compare the current version with the version from the parent commit of the entire branch, so I get all the changes at once, no matter how many commits there were.

> Also, since the Action is taking care of that, I'd say let's go for simplicity instead of speed/optimization.

I just don't use it, so I is not able to provide an exact example for the `run-clang-tidy` :)

Reply:
> I just don't use it, so I is not able to provide an exact example for the run-clang-tidy :)

I guess you are responding to a different comment, but just mention `run-clang-tidy` as an option (in the text before the sample yaml file) so the readers can look it up themselves if they need to.

Reply:
I feel that I have chosen the wrong wording regarding explanation of how the `clang-tidy-diff` works in case or PRs. Suppose we have this `master` branch:

```
A - B
```

and now we are creating a branch for a future pull request:

```
A - B [master]
    \
     C - D - E [pull request]
```

when the pull request is created, the following _potential_ merge commit `M` is created:

```
A - B - - - -  M[master]
    \         /
     C - D - E [pull request]
```

and `git diff -U0 HEAD^` means comparison between `M` and `B`, so we get all `C`, `D` and `E` commits in that diff.

Reply:
Yeap, got that, cool! :) I still think however that while it's a totally fine change, it adds a degree of complexity to an example that should be simple.

Reply:
If the `git diff -U0 HEAD^ |` is there only for optimization, can you please remove it from the examples? :angel: